src/ml/neural_net/combine_mock.hpp

```diff
@@ -0,0 +1,84 @@
+/* Copyright © 2020 Apple Inc. All rights reserved.
```

Comment:
Would it be better for mocks to go somewhere under the top level `test` directory rather than the top level `src`?

Reply:
Good question. I'd like to encourage the mocks to live near the interfaces they are mocking, so that it's easier to remember that they need to change in concert. Ideally the responsibility for maintaining a mock is for the maintainer of that interface, not the clients who want to consume that interface. Since these are header-only, it's harmless to have in that directory: it doesn't get compiled into the shipping binary